cli/internal/cmd/iamcreate.go

```diff
@@ -582,3 +587,57 @@ func awsZoneToRegion(zone string) (string, error) {
 	}
 	return fmt.Sprintf("%s-%s-%c", parts[0], parts[1], parts[2][0]), nil
 }
+
+// ValidateConfigWithFlagCompatibility checks if the config is compatible with the flags.
+func ValidateConfigWithFlagCompatibility(iamProvider cloudprovider.Provider, cfg config.Config, flags iamFlags) error {
+	if !hasProviderCfg(iamProvider, cfg) {
+		return fmt.Errorf("iam provider %q seems different from config provider", iamProvider)
+	}
+
+	return checkIfCfgZoneAndFlagZoneDiffer(iamProvider, flags, cfg)
+}
+
+func checkIfCfgZoneAndFlagZoneDiffer(iamProvider cloudprovider.Provider, flags iamFlags, cfg config.Config) error {
+	flagZone := flagZoneOrAzRegion(iamProvider, flags)
+	configZone := configZoneOrAzRegion(iamProvider, cfg)
```

Comment:
```suggestion
	configZone := cfg.GetZone()
```
this one doesn't exist yet, but there's already a similar func GetRegion